contributors/devel/godep.md

```diff
@@ -1,128 +1,183 @@
+you need to restore its dependencies, too:
 
 ```sh
-# This will *not* let your local builds see packages that exist elsewhere on your system.
-export GOPATH=$KPATH
+hack/run-in-gopath.sh bash -c 'cd $GOPATH/src/example.com/go/frob && godep restore'
 ```
 
-3) Populate your new GOPATH.
+If the transitive deps collide with Kubernetes deps, you may have to manually
+resolve things.  This is where the ability to run a shell in this environment
+comes in handy:
 
 ```sh
-cd $KPATH/src/k8s.io/kubernetes
-./hack/godep-restore.sh
+hack/run-in-gopath.sh bash --norc --noprofile
 ```
 
-4) Next, you can either add a new dependency or update an existing one.
+### Updating a dep
 
-To add a new dependency is simple (if a bit slow):
+Sometimes we already have a dep, but the version of it is wrong.  Because of the
+way that `godep` and `go get` interact (badly) it's generally easiest to hit it
+with a big hammer:
 
 ```sh
-cd $KPATH/src/k8s.io/kubernetes
-DEP=example.com/path/to/dependency
-godep get $DEP/...
-# Now change code in Kubernetes to use the dependency.
-./hack/godep-save.sh
+hack/run-in-gopath.sh bash -c 'rm -rf $GOPATH/src/example.com/go/frob'
+hack/run-in-gopath.sh go get -d example.com/go/frob
+hack/run-in-gopath.sh bash -c 'git -C $GOPATH/src/example.com/go/frob checkout v2.0.0'
 ```
 
-To update an existing dependency is a bit more complicated.  Godep has an
-`update` command, but none of us can figure out how to actually make it work.
-Instead, this procedure seems to work reliably:
+This will remove the code, re-fetch it, and sync to your desired version.
+
+### Removing a dep
+
+This happens almost for free.  If you edit Kubernetes code and remove the last
+use of a given dependency, you only need to restore and save the deps, and the
+`godep` too will figure out that you don't need that dep any more:
+
+## Saving deps
+
+Now that you have made your changes - adding, updating, or removing the use of a
+dep - you need to rebuild the dependency database and make changes to the
+`vendor/` directory.
 
 ```sh
-cd $KPATH/src/k8s.io/kubernetes
-DEP=example.com/path/to/dependency
-# NB: For the next step, $DEP is assumed be the repo root.  If it is actually a
-# subdir of the repo, use the repo root here.  This is required to keep godep
-# from getting angry because `godep restore` left the tree in a "detached head"
-# state.
-rm -rf $KPATH/src/$DEP # repo root
-godep get $DEP/...
-# Change code in Kubernetes, if necessary.
-rm -rf Godeps
-rm -rf vendor
-./hack/godep-save.sh
-# Regenerate removed BUILD, licenses.
-touch vendor/BUILD
-./hack/update-bazel.sh
-./hack/update-godep-licenses.sh
-# If you haven't followed this doc step-by-step and haven't created a dedicated GOPATH,
-# make sure there is no client-go or other staging repo in $GOPATH before running the next command.
-./hack/update-staging-godeps.sh
-git checkout -- $(git status -s | grep "^ D" | awk '{print $2}' | grep ^Godeps)
+hack/run-in-gopath.sh hack/godep-save.sh
```

Comment:
Looks like we have to remove vendor/ and Godeps/ directory before running godep-save.sh

```
[dims@bigbox 15:52] ~/go/src/k8s.io/kubernetes ⟩ hack/run-in-gopath.sh hack/godep-save.sh
The directory vendor/ or Godeps/ exists. Remove them before running godep-save.sh
!!! Error in hack/run-in-gopath.sh:33
  Error in hack/run-in-gopath.sh:33. '"${@}"' exited with status 1
Call stack:
  1: hack/run-in-gopath.sh:33 main(...)
Exiting with status 1
```

Reply:
Once i clean those directory, i end up in the following failure:
https://paste.fedoraproject.org/paste/Mhy27toN7OH7YS0gkECJmg/raw